x/mint/keeper/genesis.go

```diff
@@ -6,17 +6,31 @@ import (
 	"github.com/osmosis-labs/osmosis/v7/x/mint/types"
 )
 
+const developerVestingAmount = 225_000_000_000_000
+
 // InitGenesis new mint genesis.
 func (k Keeper) InitGenesis(ctx sdk.Context, ak types.AccountKeeper, bk types.BankKeeper, data *types.GenesisState) {
+	if data == nil {
+		panic("nil mint genesis state")
+	}
+
 	data.Minter.EpochProvisions = data.Params.GenesisEpochProvisions
 	k.SetMinter(ctx, data.Minter)
 	k.SetParams(ctx, data.Params)
 
-	if !ak.HasAccount(ctx, ak.GetModuleAddress(types.ModuleName)) {
-		ak.GetModuleAccount(ctx, types.ModuleName)
-		totalDeveloperVestingCoins := sdk.NewCoin(data.Params.MintDenom, sdk.NewInt(225_000_000_000_000))
-		k.CreateDeveloperVestingModuleAccount(ctx, totalDeveloperVestingCoins)
-		bk.AddSupplyOffset(ctx, data.Params.MintDenom, sdk.NewInt(225_000_000_000_000).Neg())
+	ak.GetModuleAccount(ctx, types.ModuleName)
```

Comment:
We should be able to delete this line, your right static check would help catch this =)

Reply:
I think this is likely to be needed. The name is confusing but, basically, this method creates a module account from the module name. If we remove it, the module account will not be created.

The line that starts failing if I remove this: https://github.com/osmosis-labs/osmosis/blob/c3081a9e10e73925db9ac89345d0e2953905c10d/x/mint/keeper/genesis_test.go#L127-L129

I ported this check from this removed file:
https://github.com/osmosis-labs/osmosis/blob/d9a4f94c15c9237289408689131acb04023f3c63/x/mint/module_test.go#L27

Reply:
I will add a comment explaining why this is needed

Reply:
oh wtf is the account keeper API if the get causes state changes to be saved :/

Make sense